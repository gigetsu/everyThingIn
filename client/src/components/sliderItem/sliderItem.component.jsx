import React from 'react';
import { withRouter } from 'react-router-dom';
import {SliderImgContainer} from './sliderItem.styles'
export const SliderItem = ({
    id, imageUrl, history
}) => (
        <a onClick={() => history.push(`/details/${id}`)}>
            <SliderImgContainer src={imageUrl} />
        </a>

    );

export default withRouter(SliderItem);
