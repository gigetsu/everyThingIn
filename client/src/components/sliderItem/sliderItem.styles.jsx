import styled from 'styled-components';

export const SliderItemContainer = styled.div`
  width: 380px;
  display: flex;
  flex-direction: column;
`;

export const SliderImgContainer = styled.img`
  margin: auto;
`;

SliderImgContainer.displayName = 'SliderImgContainer';