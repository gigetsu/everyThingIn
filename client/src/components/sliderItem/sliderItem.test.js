import React from 'react';
import { shallow } from 'enzyme';

import { SliderItem } from './sliderItem.component';

describe('SliderItem component', () => {
    let wrapper;
    let mockHistory;
    const mockId = 'testId'
    const imageUrl = 'testImage';
  
    beforeEach(() => {
  
      mockHistory = {
        push: jest.fn()
      };
  
      const mockProps = {
        id: mockId,
        history: mockHistory,
        imageUrl
      };
  
      wrapper = shallow(<SliderItem {...mockProps} />);
    });
  
    it('should render SliderItem component', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should call history.push with the right string when a tag clicked', () => {
      wrapper.find('a').simulate('click');
  
      expect(mockHistory.push).toHaveBeenCalledWith(`/details/${mockId}`);
    });
  
    it('should pass imageUrl to SliderImgContainer as the prop imageUrl', () => {
      expect(wrapper.find('SliderImgContainer').prop('src')).toBe(
        imageUrl
      );
    });
  });