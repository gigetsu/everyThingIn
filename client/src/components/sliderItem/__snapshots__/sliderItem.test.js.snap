// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderItem component should render SliderItem component 1`] = `ShallowWrapper {}`;
